src/transformers/models/llama4/configuration_llama4.py

```
@@ -125,6 +125,9 @@ def __init__(
         self.vision_feature_layer = vision_feature_layer
         self.vision_feature_select_strategy = vision_feature_select_strategy
         self.rope_theta = rope_theta
+
+        if not self.vision_feature_layer:
+            logger.warning("`vision_feature_layer` is deprecated and will be removed in v4.58")
```

i think we just need warning when users try to get the attribute or set to a different value, letting them know that `vision_feature_layer` has no effect on model

---

@zucchini-nlp A code suggestion or reference would be better here. I can only think of a niave way to have getter and setter funcs for vision_feature_layer attr and raise a warning like you mentioned (instead of general warning) or is there a util func or decorator which can be used ootb